### PR TITLE
[DT-355][risk=no] Fix update-cloud-config command

### PR DESCRIPTION
**Pull request by chenchals**
Description:
---
- Updated update-cloud-config to fix credential issue by wrapping in ServiceAccountContext
- This command is needed to update featured workspaces / demos / tutorials for the upcoming CDR release.

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have added explanatory comments where the logic is not obvious
- [ ] I have run and tested this change locally, and my testing process is described here
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later
- [ ] If this includes an API change, I have run the E2E tests on this change against my local server with [yarn test-local](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples) because this PR won't be covered by the CircleCI tests 
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P)
- [ ] If this change impacts deployment safety (e.g. removing/altering APIs which are in use) I have documented these in the description
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and updated the appropriate API consumers
  * AoU UI
  * [Perf tests](https://github.com/broadinstitute/mcnulty/blob/develop/src/test/scala/services/AoU.scala)
  * [Researcher Directory export](https://github.com/all-of-us/workbench/wiki/Researcher-Directory-(RDR-export))
  * Cron tasks - for Offline*Controllers
  * SumoLogic - for EgressAlert

### CODE REVIEW

File: api/libproject/devstart.rb [hunk 1]
```diff
@@ -2451,7 +2451,9 @@ def update_cloud_config(cmd_name, args)
   op.parse.validate
   gcc.validate()
   ENV.update(read_db_vars(gcc))
-  load_config(gcc.project)
+  ServiceAccountContext.new(gcc.project).run do
```
Comment (jmthibault79) [inline api/libproject/devstart.rb:2454]: I'm not sure what the right thing to do is here.

This is one possibility: it's the same change as #7571 .

However, this is wrapping a different command `load_config()`, and we fixed a different instance in a different way: #7566

Comment (chenchals) [reply api/libproject/devstart.rb:2454]: ---
Tested main branch and it works for `./project.rb update-cloud-config --project all-of-us-workbench-test`
```
2023-04-12 16:43:59.252  INFO 3244 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Bootstrapping Spring Data JPA repositories in DEFAULT mode.
2023-04-12 16:43:59.357  INFO 3244 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Finished Spring Data repository scanning in 105ms. Found 32 JPA repository interfaces.
2023-04-12 16:43:59.697  INFO 3244 --- [           main] org.pmiops.workbench.db.Params           : Workbench SQL instance params: [hostname:null cloudSqlInstanceName:all-of-us-workbench-test:us-central1:workbenchmaindb username:workbench password:XX*****************]
2023-04-12 16:43:59.721  INFO 3244 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Starting...
2023-04-12 16:43:59.826  INFO 3244 --- [           main] c.g.cloud.sql.core.CoreSocketFactory     : Connecting to Cloud SQL instance [all-of-us-workbench-test:us-central1:workbenchmaindb] via SSL socket.
2023-04-12 16:43:59.826  INFO 3244 --- [           main] c.g.cloud.sql.core.CoreSocketFactory     : First Cloud SQL connection, generating RSA key pair.
2023-04-12 16:44:01.437  INFO 3244 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Start completed.
2023-04-12 16:44:01.541  INFO 3244 --- [onnection adder] c.g.cloud.sql.core.CoreSocketFactory     : Connecting to Cloud SQL instance [all-of-us-workbench-test:us-central1:workbenchmaindb] via SSL socket.
2023-04-12 16:44:01.553  INFO 3244 --- [           main] o.hibernate.jpa.internal.util.LogHelper  : HHH000204: Processing PersistenceUnitInfo [name: workbench]

```
with main branch:  `./project.rb update-cloud-config --project all-of-us-rw-preprod` <- **preprod - fails**
```
2023-04-12 16:46:20.623  INFO 3397 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Bootstrapping Spring Data JPA repositories in DEFAULT mode.
2023-04-12 16:46:20.729  INFO 3397 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Finished Spring Data repository scanning in 106ms. Found 32 JPA repository interfaces.
2023-04-12 16:46:21.076  INFO 3397 --- [           main] org.pmiops.workbench.db.Params           : Workbench SQL instance params: [hostname:null cloudSqlInstanceName:all-of-us-rw-preprod:us-central1:workbenchmaindb username:workbench password:xo******************]
2023-04-12 16:46:21.101  INFO 3397 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Starting...
2023-04-12 16:46:21.210  INFO 3397 --- [           main] c.g.cloud.sql.core.CoreSocketFactory     : Connecting to Cloud SQL instance [all-of-us-rw-preprod:us-central1:workbenchmaindb] via SSL socket.
2023-04-12 16:46:21.210  INFO 3397 --- [           main] c.g.cloud.sql.core.CoreSocketFactory     : First Cloud SQL connection, generating RSA key pair.
2023-04-12 16:46:22.260 ERROR 3397 --- [pool-1-thread-2] c.g.cloud.sql.core.CloudSqlInstance      : Got more than one input failure. Logging failures after the first

java.lang.RuntimeException: [all-of-us-rw-preprod:us-central1:workbenchmaindb] The Cloud SQL Instance does not exist or your account is not authorized to access it. Please verify the instance connection name and check the IAM permissions for project "all-of-us-rw-preprod" 
        at com.google.cloud.sql.core.CloudSqlInstance.addExceptionContext(CloudSqlInstance.java:621) ~[jdbc-socket-factory-core-1.7.0.jar:na]
        at com.google.cloud.sql.core.CloudSqlInstance.fetchMetadata(CloudSqlInstance.java:510) ~[jdbc-socket-factory-core-1.7.0.jar:na]

```
---
With branch chenchals/DT-355 (for this PR) `./project.rb update-cloud-config --project all-of-us-workbench-test`
```

2023-04-12 16:39:57.056  INFO 2571 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Bootstrapping Spring Data JPA repositories in DEFAULT mode.
2023-04-12 16:39:57.159  INFO 2571 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Finished Spring Data repository scanning in 102ms. Found 32 JPA repository interfaces.
2023-04-12 16:39:57.509  INFO 2571 --- [           main] org.pmiops.workbench.db.Params           : Workbench SQL instance params: [hostname:null cloudSqlInstanceName:all-of-us-workbench-test:us-central1:workbenchmaindb username:workbench password:p5*****************]
2023-04-12 16:39:57.530  INFO 2571 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Starting...
2023-04-12 16:39:57.622  INFO 2571 --- [           main] c.g.cloud.sql.core.CoreSocketFactory     : Connecting to Cloud SQL instance [all-of-us-workbench-test:us-central1:workbenchmaindb] via SSL socket.
2023-04-12 16:39:57.622  INFO 2571 --- [           main] c.g.cloud.sql.core.CoreSocketFactory     : First Cloud SQL connection, generating RSA key pair.
2023-04-12 16:39:59.136  INFO 2571 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Start completed.
2023-04-12 16:39:59.220  INFO 2571 --- [           main] o.hibernate.jpa.internal.util.LogHelper  : HHH000204: Processing PersistenceUnitInfo [name: workbench]

```
With branch chenchals/DT-355 (for this PR) `./project.rb update-cloud-config --project all-of-us-rw-preprod` **Preprod works-as expected**
```
2023-04-12 16:38:52.200  INFO 2405 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Finished Spring Data repository scanning in 188ms. Found 32 JPA repository interfaces.
2023-04-12 16:38:52.212  INFO 2405 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Bootstrapping Spring Data JPA repositories in DEFAULT mode.
2023-04-12 16:38:52.325  INFO 2405 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Finished Spring Data repository scanning in 112ms. Found 32 JPA repository interfaces.
2023-04-12 16:38:52.691  INFO 2405 --- [           main] org.pmiops.workbench.db.Params           : Workbench SQL instance params: [hostname:null cloudSqlInstanceName:all-of-us-rw-preprod:us-central1:workbenchmaindb username:workbench password:XX******************]
2023-04-12 16:38:52.716  INFO 2405 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Starting...
2023-04-12 16:38:52.824  INFO 2405 --- [           main] c.g.cloud.sql.core.CoreSocketFactory     : Connecting to Cloud SQL instance [all-of-us-rw-preprod:us-central1:workbenchmaindb] via SSL socket.
2023-04-12 16:38:52.824  INFO 2405 --- [           main] c.g.cloud.sql.core.CoreSocketFactory     : First Cloud SQL connection, generating RSA key pair.
2023-04-12 16:38:54.344  INFO 2405 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Start completed.
2023-04-12 16:38:54.449  INFO 2405 --- [           main] o.hibernate.jpa.internal.util.LogHelper  : HHH000204: Processing PersistenceUnitInfo [name: workbench]

```

**Review by dmohs — APPROVED** [review]
Whatever works is fine with me.

**Review by jmthibault79 — APPROVED** [review]
Great testing, @chenchals - LGTM

**Pull request merged**